datum/docker-compose.yml

```diff
@@ -0,0 +1,11 @@
+version: "3.7"
+
+services:
+  datum:
+    image: ghcr.io/retropex/datum:v1.3
+    volumes:
+      - type: bind
+        source: ${APP_DATA_DIR}/settings/datum_gateway_config.json
+        target: /app/datum_gateway_config.json
+    ports:
+      - 21000:21000
```

Comment:
Instead of defining a port mapping here, what we can do is add our app proxy container as a service in this compose file. And then point the proxy at the `datum` container.

See your Bitcoin Knots app as an example:
https://github.com/getumbrel/umbrel-apps/blob/d33eef2391a2896194275298b9b884af924f738c/bitcoin-knots/docker-compose.yml#L4-L7

So you'd do this:
```yml
app_proxy:
    environment:
      APP_HOST: datum_datum_1
      APP_PORT: 21000
```
That way Datum is running behind our transparent proxy and inherits the security properties of our umbrel auth:

- By default, we only let requests through the proxy if the user already has a valid auth cookie from the Umbrel homescreen. So if datum has no auth of its own, it is protected by ours.
- It also has the benefit of inheriting other security properties of Umbrel auth, such as 2FA if they have it enabled on their Umbrel. They would then get 2FA security for all apps behind the auth proxy too
- And then if you need to allow external connections to datum running on port 21000 (e.g., from something not accessing the UI), we can whitelist certain routes, so that the web UI is protected by auth, but something like `/connect-to-datum/*` isn't:

e.g., adding this env var to the proxy container
```
PROXY_AUTH_WHITELIST: "/api/*"
```

- You can also disable the auth portion of the app proxy container entirely with this env var:
```yaml
PROXY_AUTH_ADD: "false"
```

Would something like this be compatible with how datum will work?

Reply:
Good idea, the only port who need to be freely available for ASIC's is the port `23334`, should I expose it with the `docker-compose.yml`

Reply:
also if I add the following block to `docker-compose.yml` should I remove the port under datum section?
```
app_proxy:
    environment:
      APP_HOST: datum_datum_1
      APP_PORT: 21000
```

Reply:
Yep, that's exactly right.
- add the 23334 port binding to the `docker-compose.yml` if that is the port that the asic will connect on
- remove the 21000 port binding under the `datum` service